lms/static/js/learner_dashboard/views/course_enroll_view.js

```diff
@@ -20,23 +20,89 @@
                 tpl: HtmlUtils.template(pageTpl),
 
                 events: {
-                    'click .enroll-button': 'handleEnroll'
+                    'click .enroll-button': 'handleEnroll',
+                    'change .run-select': 'handleRunSelect',
                 },
 
                 initialize: function(options) {
-                    if (options.$el){
-                        this.$el = options.$el;
-                        this.render();
+                    this.$parentEl = options.$parentEl;
+                    this.enrollModel = options.enrollModel;
+                    this.urlModel = options.urlModel;
+                    this.render();
+                    if(this.urlModel){
+                        this.trackSelectionUrl = this.urlModel.get('track_selection_url');
                     }
                 },
 
                 render: function() {
-                    var filledTemplate = this.tpl(this.model.toJSON());
-                    HtmlUtils.setHtml(this.$el, filledTemplate);
+                    var filledTemplate;
+                    if (this.$parentEl &&
+                        this.enrollModel &&
+                        this.model.get('course_key')){
+
+                        filledTemplate = this.tpl(this.model.toJSON());
+                        HtmlUtils.setHtml(this.$el, filledTemplate);
+                        HtmlUtils.setHtml(this.$parentEl, HtmlUtils.HTML(this.$el));
+                    }
                 },
 
                 handleEnroll: function(){
                     //Enrollment click event handled here
+                    if (!this.model.get('is_enrolled')){
+                        // actually enroll
+                        this.enrollModel.save({
+                            course_id: this.model.get('course_key')
+                        }, {
+                            success: _.bind(this.enrollSuccess, this),
+                            error: _.bind(this.enrollError, this)
+                        });
+                    }
+                },
+
+                handleRunSelect: function(event){
+                    var runKey;
+                    if (event.target){
+                        runKey = $(event.target).val();
+                        if (runKey){
+                            this.model.updateRun(runKey);
+                        }
+                    }
+                },
+
+                enrollSuccess: function(){
+                    var courseKey = this.model.get('course_key');
+                    if (this.trackSelectionUrl) {
+                        // Go to track selection page
+                        this.redirect( this.trackSelectionUrl + courseKey );
+                    }else{
```

Comment:
I don't think this branch will ever be hit. Is this here just in case the track selection URL is missing?